### PR TITLE
13 option to save model with lowest cross val loss

### PR DESCRIPTION
**Pull request by abuzarmahmood**
- **feat: enhance model training to save the best model based on cross-validation loss**
- **fix: resolve undefined names for plot_dir, model_name, and plt in run_model.py**

### CODE REVIEW

**Comment by abuzarmahmood** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/abuzarmahmood/blechRNN/14?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#14** <a href="https://app.graphite.dev/github/pr/abuzarmahmood/blechRNN/14?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/abuzarmahmood/blechRNN/14?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#15** <a href="https://app.graphite.dev/github/pr/abuzarmahmood/blechRNN/15?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
